### PR TITLE
[WIP] IMAP support

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -223,3 +223,6 @@ end
 gem 'webpush', require: false
 gem 'colored2', require: false
 gem 'maxminddb'
+
+gem 'google-api-client'
```
Comment (SamSaffron) [inline Gemfile]: got to require false this

File: Gemfile [hunk 2]
```diff
@@ -223,3 +223,6 @@ end
...
+gem 'googleauth'
```
Comment (SamSaffron) [inline Gemfile]: same for this require false

File: Gemfile.lock [hunk 3]
```diff
@@ -177,11 +195,15 @@ GEM
     mail (2.7.1)
       mini_mime (>= 0.1.1)
     maxminddb (0.1.22)
+    memoist (0.16.0)
     memory_profiler (0.9.13)
     message_bus (2.2.0)
       rack (>= 1.1.3)
     metaclass (0.0.4)
     method_source (0.9.2)
+    mime-types (3.2.2)
```
Comment (SamSaffron) [inline Gemfile.lock]: oh my... this is going to be a problem ... we can not add this

File: Gemfile.lock [hunk 4]
```diff
@@ -378,6 +406,11 @@ GEM
       rack (>= 1.5.0)
       rack-protection (>= 1.5.0)
       redis (>= 3.3.5, < 5)
+    signet (0.11.0)
+      addressable (~> 2.3)
```
Comment (tgxworld) [inline Gemfile.lock]: @SamSaffron  We probably don't want addressable in production as well.

**Comment by udan11** [comment]
Closing this pull request in favor of new one which has all conflicts resolved and other improvements.

**Pull request closed without merging**